### DISCUSSION TITLE
Help needed to bypass SGW

### DISCUSSION THREAD

**Discussion by alexpalcau** [cedricp/ddt4all#1207] [Q&A]
UPDATE! 
With the help of the users who answered this topic, I managed to activate android auto wireless on EasyLink 7.
Things to do:
A. MANDATORY - minimum EasyLink software version 283C36016R 
B. a good ELM v. 1.5 interface
C. SGW bypass either by connecting directly to CAN-High and LOW, or by disconnecting the Gateway plug and connecting it to a 16+32 adapter
D. Unpair all devices and pair them later.
E. Activating functions with ddt4all, following all known procedures.
Everything should go well.

Initial question:
Hello! In order to activate carplay on a renault zoe 2020, I tried two methods that led to the same negative result: I disconnected the plug from the gateway and connected the obd to it with a 16+32 adapter cable;  then I disconnected the plug from the easylink computer and connected the obd to can-high and can-low using dupont wires.  in both situations I managed to enter the ECU a-ivi, the obd read the data from the ecu but after 1-2 seconds it refreshed and all the columns showed "no data" and the application froze (not responding).  I need your opinions.

**Comment by Furtif** [comment]
maybe this post https://github.com/cedricp/ddt4all/discussions/1199

**Reply by alexpalcau** [reply]
I dont think so because when I click on the specified tab, in the first instance I can see all the data but immediately it refreshes and shows no data. I think is something regarded to the sgw but i am not sure...

**Comment by AndyIreland** [comment]
if it not a SGW issue then try changing this can timeout to 600ms and then manually refresh
![image](https://github.com/user-attachments/assets/ed577e1c-6a6b-4c10-bc43-127e139eac89)

**Comment by alexpalcau** [comment]
Well, I tried but without succes. I even installed other versions of the app. No one reported this type of issue regarding the sgw. I attach a video to show exactly what is happening: https://youtu.be/7DKspUDei4w?si=IFkIOFUqvEbxbOJl
Now, this happens even if I just plug the elm into the dedicated socket in order just to read data.

**Reply by AndyIreland** [reply]
try big green button refresh instead of blue button - i dont know if it will make difference or not but try
![image](https://github.com/user-attachments/assets/fcf1f7ec-7261-4de4-95fc-cbf5fcbd37ef)

**Reply by alexpalcau** [reply]
Tried but also failed...

**Reply by alexpalcau** [reply]
UPDATE: after the advice from @KarelSvo I took other elm (vgate) that worked perfectly! So, the problem was the chineese ELM, nothing regarded to the SGW. Btw, I can confirm that the connection through the 16+32 cable works great, no need to "steal" any CAN H/L signal from other place.
Too bad that even I activated the Wireless Android Auto function, it doesnt work on my zoe and now, even the USB connection is not working...

**Reply by AndyIreland** [reply]
what software version number is on your renault easylink radio when you go into the menu? - think I read somewhere that you had to enable wireless on the easylink at a certain software version , but after a later software update you could not enable wireless android auto. 

In am still miffed that you have managed to have a Renault ZE50 and that its not SGW protected

**Reply by matzel82** [reply]
Indeed. The latest version of Zoe have no SGW. This is the cause why it stop to manufactured soon.

**Reply by AndyIreland** [reply]
> Indeed. The latest version of Zoe have no SGW. This is the cause why it stop to manufactured soon.

OP zoe is 2020 Zoe though

**Reply by matzel82** [reply]
Yes, of course. There is a new European Directive 2019/2144. This regulates the equipment of new vehicles with prescribed security systems and systems to protect against cyber attacks. That's why the Zoe falls victim and will no longer be built and no longer newly approved. The vehicle no longer complies with the GSR2 directive. The Zoe therefore does not have a SGW.

**Reply by AndyIreland** [reply]
thanks @matzel82  , so are you saying that all Zoe (including ZE50) produced after 2019 does not have CGW implemented on the OBD2 system?

**Reply by matzel82** [reply]
At least that's how I relied on the statement of our technicians at Renault

Zoe C1A Can Gateway, vehicles from 2022 C1A HS with Security Gateway

normally you can directly access a ECU without GW. SGW no Chance…Normally… but, what is normally. Good luck

**Reply by alexpalcau** [reply]
> what software version number is on your renault easylink radio when you go into the menu? - think I read somewhere that you had to enable wireless on the easylink at a certain software version , but after a later software update you could not enable wireless android auto.
> 
> In am still miffed that you have managed to have a Renault ZE50 and that its not SGW protected

I think I manually selected A-IVI-1.11 because on the automatic search it couldnt find a specific ecu. The function is activated because I can see the WI-Fi menu on the easylink and the phone (fold6) gives a pop-up saying that the WAA is available but unfortunately, it doesnt connect... neither by usb now...

Just to be clear: my 2020 Zoe HAS SGW (located under the glovebox) and in order to bypass it, I disconnected its 32pin connector and connected it to the ELM via a specific adapter. You can bypass it also by mounting arduino pins on the easylink can-h/l.

**Reply by matzel82** [reply]
Connect by Pin-Box ??

**Reply by alexpalcau** [reply]
What do you mean by "pin-box"?

**Reply by matzel82** [reply]
> What do you mean by "pin-box"?

Like this 
![IMG_3218](https://github.com/user-attachments/assets/3cb5d095-095d-4c7c-b8dd-6534d078b28a)

Include a special wire to connect the pins/connectors

**Reply by alexpalcau** [reply]
Not really. I used something like this😅:
![20240822_173025](https://github.com/user-attachments/assets/aad2fddf-13dd-484f-a349-1e138f81c9f2)
But it is easier to use this:
![image](https://github.com/user-attachments/assets/b0ea0469-8f06-4c00-9e55-41b288095c2e)

**Reply by AndyIreland** [reply]
> > what software version number is on your renault easylink radio when you go into the menu? - think I read somewhere that you had to enable wireless on the easylink at a certain software version , but after a later software update you could not enable wireless android auto.
> > In am still miffed that you have managed to have a Renault ZE50 and that its not SGW protected
> 
> I think I manually selected A-IVI-1.11 because on the automatic search it couldnt find a specific ecu. The function is activated because I can see the WI-Fi menu on the easylink and the phone (fold6) gives a pop-up saying that the WAA is available but unfortunately, it doesnt connect... neither by usb now...
> 
> Just to be clear: my 2020 Zoe HAS SGW (located under the glovebox) and in order to bypass it, I disconnected its 32pin connector and connected it to the ELM via a specific adapter. You can bypass it also by mounting arduino pins on the easylink can-h/l.

That box located behind the glovebox I think you will find is the BCM (body control module) or the radio itself? 
Did you follow all the instructions like in this step by step?:
https://myrenaultclioblog.wordpress.com/2023/05/22/add-wireless-android-auto-wireless-car-play-to-renault-clio-mk5-bypassing-the-security-gateway/

**Reply by alexpalcau** [reply]
Regarding the sgw, it is located under the easylink ECU. The BCM is located behind the fusebox. I followed that instructions and I confirm you can bypass the sgw in that way and also another(cable adapter). I have also tried directly on the OBD port but the command doesnt reply correctly.

**Reply by AndyIreland** [reply]
> 

![image](https://github.com/user-attachments/assets/2ca0926a-31fd-4593-ac11-cd9757515458)

**Reply by bokkoman** [reply]
Does this bypass cable work? And where exactly do you connect it to?
https://private-user-images.githubusercontent.com/41481543/361249982-b0ea0469-8f06-4c00-9e55-41b288095c2e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5MTE4ODgsIm5iZiI6MTczMzkxMTU4OCwicGF0aCI6Ii80MTQ4MTU0My8zNjEyNDk5ODItYjBlYTA0NjktOGYwNi00YzAwLTllNTUtNDFiMjg4MDk1YzJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjExVDEwMDYyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk4N2ExYmM0Y2I5MzkzMDViZDQ2NGMwZWY4MTg0Yjg4ZjI0ZmRkMzY4ZmRmNTU1ZDUxZTNlY2RhNjY3NzJhOTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5pTV4iDfBsd51IWTm1vXsg64Aq0-0oheiWveHyYg1oU

**Reply by pyren4all** [reply]
> Does this bypass cable work? And where exactly do you connect it to? https://private-user-images.githubusercontent.com/41481543/361249982-b0ea0469-8f06-4c00-9e55-41b288095c2e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5MTE4ODgsIm5iZiI6MTczMzkxMTU4OCwicGF0aCI6Ii80MTQ4MTU0My8zNjEyNDk5ODItYjBlYTA0NjktOGYwNi00YzAwLTllNTUtNDFiMjg4MDk1YzJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjExVDEwMDYyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk4N2ExYmM0Y2I5MzkzMDViZDQ2NGMwZWY4MTg0Yjg4ZjI0ZmRkMzY4ZmRmNTU1ZDUxZTNlY2RhNjY3NzJhOTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5pTV4iDfBsd51IWTm1vXsg64Aq0-0oheiWveHyYg1oU

Tried it and no it doesn't. I don't know why people said it works, this cable is not adapted.

**Comment by AndyIreland** [comment]
this any help:
https://renault--zoe-forumpro-fr.translate.goog/t18076p125-guide-activer-apple-carplay-sans-fil-sur-les-zoe-52-r110-r135-bricoleurs-avertis-uniquement?_x_tr_sl=fr&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=sc

**Reply by alexpalcau** [reply]
It doesnt help. On all the forums there is only one guy who says he succesfully activated the WAA and it really works (paired with a Redmi android 11).
Although, in that forum somone asks if the "wireless mode is activated in developer mode" but I cant figure what he asks.

**Reply by AndyIreland** [reply]
yes, try going into developer mode on the easylink on your Zoe (if you can) and see if WAA and AACP is enabled. I cannot find the post again but if I do I will post it. Pretty sure they said they went into developer/hidden feature mode on the easylink by pressing 3 times on the software version on the screen of the easylink touch screen , or something like that.

**Reply by alexpalcau** [reply]
Interesting thing about the developer mode. I will search.

**Reply by AndyIreland** [reply]
its possible that it might show up in developer/hidden settings mode after a certain version update of the easylink, possibly?
Its also possible that I was reading that you can put the R-Link2 into developer mode rather than the EasyLink/EasyConnect

**Reply by alexpalcau** [reply]
Certanly, both r-link know developer mode. It is only the second time I hear about this regarding the easylink.

**Reply by alexpalcau** [reply]
> its possible that it might show up in developer/hidden settings mode after a certain version update of the easylink, possibly? Its also possible that I was reading that you can put the R-Link2 into developer mode rather than the EasyLink/EasyConnect

Tried to enter developer mode by taping the serial number without succes. Btw, i have 283C38683R software version and when clicking on update it says there is no update available

**Reply by AndyIreland** [reply]
on R-Link2 , you need to have a USB inserted into USB of car formatted to Fat32 with a LGE file on it to go into Developer Mode - I wonder if you have to do same on easylink maybe?

**Reply by AndyIreland** [reply]
> > its possible that it might show up in developer/hidden settings mode after a certain version update of the easylink, possibly? Its also possible that I was reading that you can put the R-Link2 into developer mode rather than the EasyLink/EasyConnect
> 
> Tried to enter developer mode by taping the serial number without succes. Btw, i have 283C38683R software version and when clicking on update it says there is no update available

V2.10? - and I think the latest is V2.13 according to this March 2024 
https://www.gps-rlink.com/easy-link/easy-link-software-version/

If you not been getting updates OTA I wonder if you have to have them installed at Renault garage? (sorry I dont have a ZE50 so I dont know) - maybe with updates you might achieve what you want to enable (unless its a hardware issue)

**Reply by kartoffelpflanze** [reply]
> Tried to enter developer mode by taping the serial number without succes. Btw, i have 283C38683R software version and when clicking on update it says there is no update available

I wanted to enable wireless Android Auto on my Easylink too, so I contacted someone who does these kinds of modifications. They also told me enabling it only works on a few versions:
![445880776_760621149589837_2940330894449857499_n](https://github.com/user-attachments/assets/72cd0996-f088-48ee-8203-1793a42632a1)

**Comment by alexpalcau** [comment]
UPDATE! Succeeded!
I updated the system to the *6016R version and the wireless android auto works perfect!
The next stept will be to find if it is possible to see the road indications on the dashboard as the native navigation software.

**Reply by kartoffelpflanze** [reply]
I think that requires a special video cable between the Easylink and Cluster. Maybe different hardware too, but I'm not an expert.

**Reply by kartoffelpflanze** [reply]
> I updated the system to the *6016R version

Hey, how did you do it? Mine shows no update available. You said yours was *8683R previously, and mine is *0613R, which the gps-rlink website puts in the same "category", v2.10.

**Reply by alexpalcau** [reply]
Read this and do exactly as the author writes.
https://www.drive2.ru/l/668012688575569514/

**Reply by kartoffelpflanze** [reply]
Ahh, I always forget to check this site. Many thanks.

**Reply by kartoffelpflanze** [reply]
The author says the guide lines on the parking camera disappear on every reboot. Do you have the 360 degree camera package and could you confirm if this happened to you or not?

**Reply by alexpalcau** [reply]
Unfortunately, I dont have the 360 camera package....

**Reply by Grubuntu** [reply]
>   The next stept will be to find if it is possible to see the road indications on the dashboard as the native navigation software.


I'm also very interested in this feature, feel free to share your progress if you manage to do it. I managed to enable the temperature and time display on the dashboard thanks to the information shared by Crash_Override
https://git.bit-cloud.de/Crash_Override/Zoe_PH2_DDT_Mods/src/branch/main

EDIT new link : https://git.0-c.de/Crash_Override/Zoe_PH2_DDT_Mods/src/branch/main

 but the navigation arrows display does not work for me.

**Reply by kartoffelpflanze** [reply]
I have installed the update, and, so far, cameras, guide lines and automatic parking all work the same as before. There was only one mishap where the camera didn't activate upon reverse gear, and after pressing the camera button, the 360 camera was stretched strangely. I hope this lasts.

I told someone who knows more about Easylink about what I have done, and he swore at me in a few languages, "what's going through your head installing random software on 2000 euro hardware" 😅

**Reply by alexpalcau** [reply]
@Grubuntu man, you are awesome! I didnt know about the site you indicated. With your hint, I activated on the dashboard
- clock
- temperature
- navigation arrow! which works very well with waze. I have tried with google maps but the indication is not shown
- 
![20240829_194254](https://github.com/user-attachments/assets/a603e0d0-dcd3-498c-9ec4-b5050795dda0)

**Reply by kartoffelpflanze** [reply]
@alexpalcau I don't know if by "road indications" you meant only the arrows like you have already activated. When you said that, I thought you were talking about the 3D maps. If so, here is a "tutorial": [https://www.youtube.com/watch?v=URBHOHODeOI](https://www.youtube.com/watch?v=URBHOHODeOI).
It's necessary to change the Easylink module to one with an additional video output and to add an LVDS cable between it and the dashboard.

**Reply by Mrbibi38** [reply]
@alexpalcau Hello,

You said that you succed using a good ELM v. 1.5 interface and a 16+32 adapter.

Do you have any reference to buy these? I find a lot of then but don't really know if they are working correctly. 
Thank you :)

**Reply by krik99** [reply]
> > The next stept will be to find if it is possible to see the road indications on the dashboard as the native navigation software.
> 
> I'm also very interested in this feature, feel free to share your progress if you manage to do it. I managed to enable the temperature and time display on the dashboard thanks to the information shared by Crash_Override https://git.bit-cloud.de/Crash_Override/Zoe_PH2_DDT_Mods/src/branch/main
> 
> but the navigation arrows display does not work for me.

I can't access this link in 2025 — I'm getting a DNS_PROBE_FINISHED_NXDOMAIN error. Is there a mirror available, or is it just me having this issue?

**Reply by mishu28nmv** [reply]
> "I can't access this link in 2025 — I'm getting a DNS_PROBE_FINISHED_NXDOMAIN error. Is there a mirror available, or is it just me having this issue?"
>
It's not on you. The link is not working!

**Reply by Grubuntu** [reply]
> I can't access this link in 2025 — I'm getting a DNS_PROBE_FINISHED_NXDOMAIN error. Is there a mirror available, or is it just me having this issue?
@krik99 

https://git.0-c.de/Crash_Override/Zoe_PH2_DDT_Mods/src/branch/main

**Comment by alexpalcau** [comment]
I think you may use any of those on the internet. You may take a look at:
1. https://a.aliexpress.com/_EjR7UnM
2. https://a.aliexpress.com/_EwMPE8A

mar., 23 sept. 2025, 21:39 Lilian ***@***.***> a scris:
